src/functional/ffront/foast_to_itir.py

```diff
@@ -108,59 +115,100 @@ def visit_Subscript(self, node: foast.Subscript, **kwargs) -> itir.FunCall:
         )
+        return self._lift_if_field(node)(result)
 
     def _visit_shift(self, node: foast.Call, **kwargs) -> itir.FunCall:
+        result = None
         match node.args[0]:
             case foast.Subscript(value=foast.Name(id=offset_name), index=int(offset_index)):
-                return im.shift_(offset_name, offset_index)(self.visit(node.func, **kwargs))
+                result = im.shift_(offset_name, offset_index)(self.visit(node.func, **kwargs))
             case foast.Name(id=offset_name):
-                return im.shift_(offset_name)(self.visit(node.func, **kwargs))
-        raise FieldOperatorLoweringError("Unexpected shift arguments!")
+                result = im.shift_(offset_name)(self.visit(node.func, **kwargs))
+            case _:
+                raise FieldOperatorLoweringError("Unexpected shift arguments!")
+        return result
 
     def _visit_reduce(self, node: foast.Call, **kwargs) -> itir.FunCall:
         lowering = InsideReductionLowering()
         expr = lowering.visit(node.args[0], **kwargs)
         params = list(lowering.lambda_params.items())
-        return im.lift_(
-            im.call_("reduce")(
-                im.lambda__("accum", *(param[0] for param in params))(im.plus_("accum", expr)),
-                0,
-            )
-        )(*(param[1] for param in params))
+        result = im.call_("reduce")(
+            im.lambda__("accum", *(param[0] for param in params))(im.plus_("accum", expr)),
+            0,
+        )
+        return im.lift_(result)(*(param[1] for param in params))
 
     def visit_Call(self, node: foast.Call, **kwargs) -> itir.FunCall:
         if TypeInfo(node.func.type).is_field_type:
             return self._visit_shift(node, **kwargs)
         elif node.func.id in FUN_BUILTIN_NAMES:
-            return self._visit_reduce(node, **kwargs)
-        return self._lift_lambda(node)(
-            im.call_(self.visit(node.func, **kwargs))(*self.visit(node.args, **kwargs))
-        )
+            visitor = getattr(self, f"_visit_{node.func.id}")
+            return visitor(node, **kwargs)
+        elif node.func.id in TYPE_BUILTIN_NAMES:
+            return self._visit_type_constr(node, **kwargs)
+        result = im.call_(self.visit(node.func, **kwargs))(*self.visit(node.args, **kwargs))
+        return self._lift_if_field(node)(result)
+
+    def _visit_neighbor_sum(self, node: foast.Call, **kwargs) -> itir.FunCall:
+        return self._visit_reduce(node, **kwargs)
+
+    def _visit_type_constr(self, node: foast.Call, **kwargs) -> itir.NumberLiteral:
+        if isinstance(node.args[0], foast.Constant):
+            return im.number_(node.func.id, str(node.args[0].value))
+        return self.visit(node.args[0], **kwargs)
+
+    def visit_Constant(
+        self, node: foast.Constant, **kwargs
+    ) -> itir.NumberLiteral | itir.BoolLiteral:
+        result = None
+        match node.dtype:
+            case ct.ScalarType(kind=ct.ScalarKind.FLOAT32) | "float32":
```

Comment:
This function does the opposite. SymbolType -> typed ITIR literal. This is very likely to change anyway.

Reply:
I think this is a misunderstanding. I mean the `| "float32"` resolving the string. I recognized that the IR node actually allow strings (not convinced of that, but unrelated). Since you are touching the equivalent PAST switch (https://github.com/GridTools/gt4py/pull/711/files/835fa5f255dbb2f01286b3213cdf80f3a8ae2950#diff-ba87b9fb32927e682b1d17ed55f9d5724ce1ebb6b4dde066972203b0f1e2d1a5) this should either also be added there or removed here.